code/modules/modular_computers/computers/machinery/modular_computer.dm

```diff
@@ -39,9 +39,12 @@
 	icon_state = icon_state_unpowered
 	overlays.Cut()
 
-	if(!cpu || !cpu.enabled)
-		if (!(stat & NOPOWER) || battery_powered)
-			overlays.Add(screen_icon_screensaver)
```

Comment:
That is a lot of checks, is this many really necessary?

I notice it has a if !cpu check there. Shouldn't that be enough? A newly built frame just made of steel, shouldn't have a cpu in it yet. maybe figure out why that check isnt doing the job

am i misunderstanding something?